syntax = "proto3";


enum AirportType {
  large_airport = 0;
  medium_airport = 1;
  small_aiprort= 2;
  heliport = 3;
  closed = 4;
}

message AirportRequest {
  string continent = 1;
  AirportType airport_type = 2;
}

message Location {
  float latitude = 1;
  float longitude = 2;
}

message Airport {
  string name = 1;
  string code = 2;
  Location location = 3;
  AirportType type = 4;
}

message AirportResponse {
  repeated Airport airports = 1;
}

service DataDelivery {
  rpc Airports (AirportRequest) returns (AirportResponse);
}